services/src/main/java/io/grpc/services/HealthCheckingLoadBalancerFactory.java

```diff
@@ -171,8 +172,13 @@ public void handleResolvedAddressGroups(
         List<EquivalentAddressGroup> servers, Attributes attributes) {
       Map<String, Object> serviceConfig =
           attributes.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-      String serviceName = ServiceConfigUtil.getHealthCheckedServiceName(serviceConfig);
-      helper.setHealthCheckedService(serviceName);
+      try {
+        String serviceName = ServiceConfigUtil.getHealthCheckedServiceName(serviceConfig);
+        helper.setHealthCheckedService(serviceName);
+      } catch (MalformedConfigException e) {
```

Comment:
I dont think this was right.   Exceptions in service config should be treated more carefully.  A runtimeException here could cause a channel to stop working and never recover, which is bad.


Service config needs to be able to be fully validated before replacing the existing one, incase of problems.  If an exception comes out due to user provided data, and that causes the channel to explode, then it has the same problem as replacing the existing config.   A new service config would result in an unrecoverable state.